probe-rs/src/debug/unit_info.rs

```diff
@@ -1017,9 +1017,9 @@ impl UnitInfo {
                 }
             }
             gimli::DW_TAG_structure_type => {
-                child_variable.type_name = VariableType::Struct(
-                    type_name.unwrap_or_else(|| "<unnamed struct>".to_string()),
-                );
+                let type_name = type_name.unwrap_or_else(|| "<unnamed struct>".to_string());
+                child_variable.type_name = VariableType::Struct(type_name.clone());
```

Comment:
So this is just to voice my opinion & not really a change request: I don't really like the "we need to clone this here, but we'll just inspect and throw away one of the clones" aspect of this check and I would probably have replaced the `if let` with `let ... else { unreachable!() };` instead.

Reply:
I agree. I did not find a better way ...